### DISCUSSION TITLE
Unsupported provider: Provider email could not be found

### DISCUSSION THREAD

**Discussion by fadethegap** [supabase/supabase#5278] [Questions] [answered]
The signUp() function is performing as expected, for email as well as Google and Facebook. The signIn() function works as expected for Google and Facebook as well, but when I attempt to signIn with email I get the following error:

![Screen Shot 2022-02-01 at 5 40 42 PM](https://user-images.githubusercontent.com/47059192/152063706-091ed96b-c62f-4ec9-b928-257215dd7e18.png)

I have confirmed the signIn() function is receiving the email and password.

![Screen Shot 2022-02-01 at 5 42 27 PM](https://user-images.githubusercontent.com/47059192/152063923-458e0e3d-04ec-4ab6-a3d5-4cfc0a4fac74.png)

When I submit I am returned to the application and a user object is returned (probably a dummy object per supabase security rules.).

![Screen Shot 2022-02-01 at 5 49 26 PM](https://user-images.githubusercontent.com/47059192/152064802-32339a1a-e171-4dcf-9fe5-91d00e2aa766.png)

All the email authorizations are active:

![Screen Shot 2022-02-01 at 5 44 17 PM](https://user-images.githubusercontent.com/47059192/152064137-b91b2a94-aa25-46ec-a086-05df9de8b9f3.png)

I clicked the link in the verification email:

![Screen Shot 2022-02-01 at 5 52 23 PM](https://user-images.githubusercontent.com/47059192/152064907-a8964811-e1c6-4163-a790-272d7e25fb44.png)

And was returned to the app where it logged in and returned the user object which I had enriched with the Stripe ID - so Function Hooks are firing.

![Screen Shot 2022-02-01 at 5 56 44 PM](https://user-images.githubusercontent.com/47059192/152065569-aeead96a-c1ed-4ec3-8c0a-e62bb843e097.png)

When I signOut and attempt to signIn with email, whammo!

![Screen Shot 2022-02-01 at 5 58 55 PM](https://user-images.githubusercontent.com/47059192/152065803-577a7ae8-3c8b-4d8e-ba50-14e24174ee03.png)

What am I missing?

**Comment by GaryAustin1** [comment, marked as answer]
If I read your code correctly, you drop out of your email if statement after it executes email signin and then run signIn again with loginProvider which is still "email".
Your "else" is missing...

**Reply by fadethegap** [reply]
Every time you help me @GaryAustin1, I don't know whether I'm more embarrassed or grateful. Thank you so much!
